config/config.go

```diff
@@ -36,30 +36,44 @@ var (
 type Config struct {
 	logLevel logrus.Level `mapstructure:"log_level"`
 
-	AppMode             string           `mapstructure:"app_mode"`
-	Env                 string           `mapstructure:"env"`
-	HttpServer          httpServerConfig `mapstructure:"http_server"`
-	FinishUpTime        int64            `mapstructure:"finish_up_time"`
-	CronJob             cronJobConfig    `mapstructure:"cron_job"`
-	Store               store            `mapstructure:"store"`
-	Oracle              oracle           `mapstructure:"oracle"`
-	GasPriceDomains     []string         `mapstructure:"gas_price_domains"`
-	ConversionRatePairs []string         `mapstructure:"conversion_rate_pairs"`
-	Strategy            strategyConfig   `mapstructure:"strategy"`
-	DataValidInterval   int64            `mapstructure:"data_valid_interval"`
-	Domains             map[int]Domain
-	Resources           map[string]*Resource
+	AppMode                      string                       `mapstructure:"app_mode"`
+	Env                          string                       `mapstructure:"env"`
+	HttpServer                   httpServerConfig             `mapstructure:"http_server"`
+	FinishUpTime                 int64                        `mapstructure:"finish_up_time"`
+	CronJob                      cronJobConfig                `mapstructure:"cron_job"`
+	Store                        store                        `mapstructure:"store"`
+	ConversionRateOracles        []Oracle                     `mapstructure:"conversion_rate_oracles"`
+	GasPriceOracles              gasPriceOraclesWithDomainIDs `mapstructure:"gas_price_oracles_with_domain_ids"`
+	ConversionRatePairs          []string                     `mapstructure:"conversion_rate_pairs"`
+	Strategy                     strategyConfig               `mapstructure:"strategy"`
+	DataValidInterval            int64                        `mapstructure:"data_valid_interval"`
+	Domains                      map[int]Domain
+	Resources                    map[string]*Resource
+	GasPriceOraclesWithDomainIDs map[int][]Oracle
+}
+
+type OracleDetails struct {
+	Enable bool    `mapstructure:"enable"`
+	ApiKey string  `mapstructure:"api_key"`
+	Apis   apiUrls `mapstructure:"apis"`
 }
 
 type strategyConfig struct {
 	Local string `mapstructure:"local"`
 }
 
-type oracle struct {
-	Etherscan     etherscan     `mapstructure:"etherscan"`
-	Polygonscan   polygonscan   `mapstructure:"polygonscan"`
-	CoinMarketCap coinMarketCap `mapstructure:"coinmarketcap"`
-	Moonscan      coinMarketCap `mapstructure:"moonscan"`
+type gasPriceOraclesWithDomainIDs struct {
```

Comment:
?

Reply:
Why are we hardcoding domains here?

Reply:
this is what I said the yaml limitation, I have an idea and im working on it now, give me a moment

Reply:
yes, I know, but didnt think to use here, giveme a moment